### PR TITLE
Don't wrap JSX elements as attribute in ()

### LINKED ISSUE #3626: Long JSX elements as attribute values have invalid output
**Prettier 1.9.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeAfAHSgAh6gMQgmzz0lgRgF5UATASwDctcy97mcB6V91boxakerEYWIjy0eLFp92+IaKn5eI9VAHoQAGhAQADjAbQAzslABDAE42IAdwAKthBZRWmEBnT0grZjDIAGZWADZmcPoARjZWYADWcDAAyobxDFAA5sgwNgCuUSB0EGAh4ZH6mZE2ME5xWQC2VuURRQBWZgAeAEJxickpVo1wADKZcK2VBvkwhrMATLkFRek2Ncgg0VbRAJ5h0H6GNpkwAOo+MAAWyAAcAAz6xxCRZ3GGm8dwNUyT+jZwACO+QYAPqViaLSQoTa+kijQYy0KcMyWTCcAAivkIPApkUYDsLnRrsgFvo8lYGGFUQBhCCNZqbKDQP4gfKRAAqO3cMMiAF8+UA)

**Input:**
```jsx
<>
  <Foo
    content=<div>
      <div />
    </div>
  />

  <Foo
    content=<>
      <div />
    </>
  />
</>
```

**Output:**
```jsx
<>
  <Foo
    content=(
      <div>
        <div />
      </div>
    )
  />

  <Foo
    content=(
      <>
        <div />
      </>
    )
  />
</>;

```

**Second Output:**
```
SyntaxError: JSX value should be either an expression or a quoted JSX text (3:13)
  1 | <>
  2 |   <Foo
> 3 |     content=(
    |             ^
  4 |       <div>
  5 |         <div />
  6 |       </div>

SyntaxError: JSX value should be either an expression or a quoted JSX text (11:13)
   9 | 
  10 |   <Foo
> 11 |     content=(
     |             ^
  12 |       <>
  13 |         <div />
  14 |       </>
```

**Expected behavior:**
Same as input?

I’m honestly surprised that this is valid. However, here’s [the spec](http://facebook.github.io/jsx/#syntax):

> JSXAttributeValue:
> 
> - `"` JSXDoubleStringCharacters<sub>opt</sub> `"`
> - `'` JSXSingleStringCharacters<sub>opt</sub> `'`
> - `{` AssignmentExpression `}`
> - **JSXElement**
> - **JSXFragment**

<sup>(emphasis mine)</sup>

Ref https://github.com/facebook/jsx/issues/53 (issue about removing this) and https://github.com/facebook/jsx/pull/15 (PR adding this). It’s currently [broken](http://babeljs.io/repl#?babili=false&browsers=&build=&builtIns=false&code_lz=DwMQ9mAEBGCGBOBeYAhWAvSB6AfNnQA&debug=false&circleciRepo=&evaluate=false&lineWrap=false&presets=react&prettier=false&targets=&version=6.26.0) in Babel.

Maybe we should just change it to the normal `{}` form, which is semantically equivalent as far as I can tell and looks less awful.

### PR DESCRIPTION
**Pull request by duailibe**
Fixes #3626

### CODE REVIEW

File: tests/jsx-attr-element/attr-element.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+<Foo prop=<Bar><Baz /></Bar> />;
+<Foo prop=<><Bar><Baz /></Bar></> />;
```
Comment (azz) [inline tests/jsx-attr-element/attr-element.js]: Wow that's incredibly cryptic...

**Comment by julien-f** [comment]
It's indeed not very easy to read and the formatting is a bit different than other JSX, because the opening tag is not on its own line, maybe we should add braces?

```js
<Foo
  content={
    <div>
      <div />
    </div>
  }
/>
```

**Comment by j-f1** [comment]
I had a PR that did that @julien-f, but one of the goals of Prettier is to avoid modifying the AST, which would happen if the braces were added.

**Comment by duailibe** [comment]
@julien-f1 you can just add the braces abd Prettier will respect that. You don’t need to use the “no-braces” style

**Pull request merged**